detection_rules/rule.py

```diff
@@ -1030,6 +1033,15 @@ def _convert_get_setup_content(self, note_tree: list) -> str:
 
         return "".join(setup).strip()
 
+    def _default_bbr_fields(self, obj: dict) -> None:
```

Comment:
Rather than setting these as build time values, it is a safer approach to validate expected values, with exceptions as needed.

easiest way to do this would be to add a marshmallow validation method in `BaseRuleData`

Reply:
pseudocode would be:

```python
if not bypass_bbr_lookback:
    if not from and interval:
        invalidate
    if from != now-119m or interval != 60m:
        invalidate
```

Reply:
To follow the convention of the note checking bypass, I added this function:
https://github.com/elastic/detection-rules/blob/8d00c6a6840163186b3e7c1ed3c65c13e44cfc94/detection_rules/rule.py#L372

Initially I tried to make it a `@cached_property` of `BaseRuleData` more similar to how the note bypass is a `@cached_property` of `DataValidator`. However, since the class/the value of `self` can be multiple classes for building block rules, the property was often inaccessible in this function. Is this expected, and if not do you know of a better way to pass the property forward through the respective rule data classes? Thanks! 

<details><summary>Cached Property Attempt Details</summary>
<p>

Code:
```python
class BaseRuleData(MarshmallowDataclassMixin, StackCompatMixin):
...

    @cached_property
    def skip_validate_bbr(self) -> bool:
        return os.environ.get('DR_BYPASS_BBR_VALIDATION') is not None
...
```

Stopped at breakpoint:
![image](https://github.com/elastic/detection-rules/assets/119343520/a45ecfcd-7dbb-4615-9919-dfa4fb82ba4a)

In debugging console:
```py
self.skip_validate_bbr
Traceback (most recent call last):
  File "<string>", line 1, in <module>
AttributeError: 'QueryRuleData' object has no attribute 'skip_validate_bbr'
```

</p>
</details>

Reply:
Moved all of the functionality to `DataValidator` class, which addresses the above issue.